frigate/http.py

```diff
@@ -1156,27 +1157,50 @@ def latest_frame(camera_name):
     resize_quality = request.args.get("quality", default=70, type=int)
 
     if camera_name in current_app.frigate_config.cameras:
-        frame = current_app.detected_frames_processor.get_current_frame(
+        frame, frame_time = current_app.detected_frames_processor.get_current_frame(
             camera_name, draw_options
         )
         retry_interval = float(
             current_app.frigate_config.cameras.get(camera_name).ffmpeg.retry_interval
             or 10
         )
 
-        if frame is None or datetime.now().timestamp() > (
-            current_app.detected_frames_processor.get_current_frame_time(camera_name)
-            + retry_interval
-        ):
-            if current_app.camera_error_image is None:
-                error_image = glob.glob("/opt/frigate/frigate/images/camera-error.jpg")
+        latest_frame = (
+            frame_time + retry_interval * 1.5
+        )  # magic number, because of shit happens
+        now = datetime.now().timestamp()
+        if frame is None or now > latest_frame:
+            if (
+                now - latest_frame > 31536000
+            ):  # 1 year = 365 days * 24 hours * 60 minutes * 60 seconds
+                if current_app.camera_waiting_image is None:
+                    waiting_image = glob.glob(
+                        "/opt/frigate/frigate/images/camera-waiting.jpg"
+                    )
 
-                if len(error_image) > 0:
-                    current_app.camera_error_image = cv2.imread(
-                        error_image[0], cv2.IMREAD_UNCHANGED
+                    if len(waiting_image) > 0:
+                        current_app.camera_waiting_image = cv2.imread(
+                            waiting_image[0], cv2.IMREAD_UNCHANGED
+                        )
+                frame = current_app.camera_waiting_image
+                logger.debug(
+                    f"Return waiting image for camera {camera_name}: latency is {now - latest_frame}s, retry_interval: {retry_interval}s"
+                )
+            else:
+                if current_app.camera_error_image is None:
+                    error_image = glob.glob(
+                        "/opt/frigate/frigate/images/camera-error.jpg"
                     )
 
-            frame = current_app.camera_error_image
+                    if len(error_image) > 0:
+                        current_app.camera_error_image = cv2.imread(
+                            error_image[0], cv2.IMREAD_UNCHANGED
+                        )
+
+                frame = current_app.camera_error_image
+                logger.warning(
```

Comment:
I don't think we want a log here for this, there will already be actual errors in the logs for the camera and this will just add clutter to the logs